lib/plug/router.ex

```diff
@@ -121,9 +121,9 @@ defmodule Plug.Router do
 
   ## Routes compilation
 
-  All routes are compiled to a match function that receives
-  three arguments: the method, the request path split on `/`
-  and the connection. Consider this example:
+  All routes are compiled to a do_match function that receives
```

Comment:
np. It's really fun to read plug codes, and I learn a lot. :)

Yes, I agree it's a implementation detail. By `a mixture of both approaches`, did you mean that we fake a compiled result function that would be understandable by users reading the doc?

If so, I could totally understand that and I would close this pr. :)